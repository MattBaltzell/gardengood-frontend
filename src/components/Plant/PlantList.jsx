import "./Plant.css";

const PlantList = ({ plants }) => {
  return (
    <div className="Plant-list">
      {plants.map((plant) => {
        return (
          <div key={plant.id} className="Plant-list_thumbnail">
            <img
              className="Plant-list_img"
              alt={plant.name}
              src={plant.imgUrl}
            />
            <h3 className="Plant-list_name">{plant.name}</h3>
          </div>
        );
      })}
    </div>
  );
};

export default PlantList;
