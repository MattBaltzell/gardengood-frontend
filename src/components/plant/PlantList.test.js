import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import PlantList from "./PlantList";
import UserContext from "../auth/UserContext";

const user = {
  username: "testuser",
  first_name: "firstname",
  last_name: "lastname",
  email: "test@test.com",
};

const plants = [
  { id: 1, name: "Plant 1", imgUrl: "http://img1url.jpg" },
  { id: 2, name: "Plant 2", imgUrl: "http://img2url.jpg" },
];

describe("PlantList component tests", function () {
  it("renders without crashing", function () {
    render(
      <MemoryRouter>
        <PlantList plants={plants} />
      </MemoryRouter>
    );
  });

  it("matches snapshot", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <UserContext.Provider value={user}>
          <PlantList plants={plants} />
        </UserContext.Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders list of plants", function () {
    render(
      <MemoryRouter>
        <UserContext.Provider value={user}>
          <PlantList plants={plants} />
        </UserContext.Provider>
      </MemoryRouter>
    );
    expect(screen.getByText("Plant 1")).toBeInTheDocument();
    expect(screen.getByAltText("Plant 2")).toBeInTheDocument();
  });
});
