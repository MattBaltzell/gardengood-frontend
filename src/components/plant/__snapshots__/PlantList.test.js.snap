// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlantList component tests matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Plant-list"
  >
    <div
      class="Plant-list__thumbnail"
    >
      <img
        alt="Plant 1"
        class="Plant-list__img"
        src="http://img1url.jpg"
      />
      <h3
        class="Plant-list__name"
      >
        Plant 1
      </h3>
    </div>
    <div
      class="Plant-list__thumbnail"
    >
      <img
        alt="Plant 2"
        class="Plant-list__img"
        src="http://img2url.jpg"
      />
      <h3
        class="Plant-list__name"
      >
        Plant 2
      </h3>
    </div>
  </div>
</DocumentFragment>
`;
